platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/MapRenderer.java

```diff
@@ -121,12 +131,23 @@ private native void nativeInitialize(MapRenderer self,
   private void updateFps() {
     frames++;
     long currentTime = System.nanoTime();
-    double fps = 0;
-    if (currentTime - timeElapsed >= 1) {
-      fps = frames / ((currentTime - timeElapsed) / 1E9);
-      onFpsChangedListener.onFpsChanged(fps);
-      timeElapsed = currentTime;
-      frames = 0;
+    double fps = frames / ((currentTime - timeElapsed) / 1E9);
```

Comment:
Right hand side can be `1E9 / (currentTime - timeElapsed)` and the `frames` field can be dropped entirely as it's always going to be equal to `1`.